### ISSUE TITLE
Upload rockstor config fails

### ISSUE THREAD

**Issue by kanecko** [rockstor/rockstor-core#2483]
Original forum thread: https://forum.rockstor.com/t/upload-rockstor-config-fails/8638

My current version: Leap 15.4 V4.5.5-0.
My previous version: Leap 15.3 V4.(the latest)

Initially I installed the Leap 15.3 V4.1.0-0 and configured the system a little bit (raid1, scheduled scrubbing, two Rockons, NTP server).
Then I subscribed to the testing channel and updated fully.

When the new RC2 iso came out, I have made a Rockstor backup and re-installed/overwritten the system disk with the RC2.
Installation went fine. I rebooted afterwards.

The problem now is that I cannot upload my backup!
[backup-2023-01-22-135322.json.gz](https://github.com/rockstor/rockstor-core/files/10487482/backup-2023-01-22-135322.json.gz)

The errors are a slightly different from mobile to PC browsers, but the best clue I've got was trying to upload from PC Chrome: 
![image](https://user-images.githubusercontent.com/5829467/214240578-2715c80d-8abd-41ff-a815-d8e6d59e2abb.png)

When investigating deeper we see it fails at `POST https://192.168.0.103/api/config-backup/file-upload` with status code 400.
the "filename" seems actually defined here: `Content-Disposition: form-data; name=\"file\"; filename=\"backup-2023-01-22-135322.json.gz\"`

Frontend stacktrace: 
![image](https://user-images.githubusercontent.com/5829467/214242574-1364daaf-b1c5-4df0-abe6-61e8a4283c6b.png)

Full HAR: 
[rq.har.gz](https://github.com/rockstor/rockstor-core/files/10487539/rq.har.gz)

**Comment by FroggyFlox** [comment]
Hi @kanecko ,

Thanks for the detailed issue and the logs!
I can reproduce part of the issue and have someone that seems to fix it. If you don't mind, would you be ok with testing it? It would be a simple change that can be easily reversed. I unfortunately won't have time for more details for a few hours but let us know if you're game for testing that simple change.

**Comment by kanecko** [comment]
I'm able and willing :)

**Comment by FroggyFlox** [comment]
Thanks a lot, @kanecko !

I haven't had time to test the `sed` command below, but it should be OK. Could you please run the following?
```
sed -i 's:parser_classes = (FileUploadParser, MultiPartParser):parser_classes = [MultiPartParser]:g' /opt/rockstor/src/rockstor/storageadmin/views/config_backup.py
```
This should lead to lines 628-629 of `/opt/rockstor/src/rockstor/storageadmin/views/config_backup.py` to read:
```python
class ConfigBackupUpload(ConfigBackupMixin, rfc.GenericView):
    parser_classes = [MultiPartParser]
```
instead of:
```python
class ConfigBackupUpload(ConfigBackupMixin, rfc.GenericView):
    parser_classes = (FileUploadParser, MultiPartParser)
```

Then:
1. `systemctl stop rockstor`
2. `systemctl start rockstor`
3. Go to System > Config Backups
4. Create a new backup
5. Download this newly-created backup to your local client's disk
6. Make sure to fully refresh page (Ctrl + Shift + R in Firefox) in case there's some remnant code that would need it
7. Delete this newly-created backup in Rockstor's webUI by clicking on the "trashcan" icon
8. Try uploading the backup downloaded in step 5 above, without having touched it to make there's no interference.

Report back... please :-)

**Comment by kanecko** [comment]
works!
i've submitted a PR: #2484

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2484
